src/nv_ingest/modules/storages/embedding_storage.py

```diff
@@ -0,0 +1,159 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+import logging
+import os
+import uuid
+import traceback
+from io import BytesIO
+from typing import Any
+from typing import Dict
+from urllib.parse import quote
+
+import mrc
+import mrc.core.operators as ops
+import pandas as pd
+from minio import Minio
+from morpheus.messages import ControlMessage
+from morpheus.messages import MessageMeta
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+
+import cudf
+
+from nv_ingest.schemas.embedding_storage_schema import EmbeddingStorageModuleSchema
+from nv_ingest.schemas.metadata_schema import ContentTypeEnum
+from nv_ingest.util.exception_handlers.decorators import nv_ingest_node_failure_context_manager
+from nv_ingest.util.flow_control import filter_by_task
+from nv_ingest.util.modules.config_validator import fetch_and_validate_module_config
+from nv_ingest.util.tracing import traceable
+
+logger = logging.getLogger(__name__)
+
+MODULE_NAME = "embedding_storage"
+MODULE_NAMESPACE = "nv_ingest"
+
+# TODO: Move these into pipeline.py to populate the stage and validate them using the pydantic schema on startup.
+_DEFAULT_ENDPOINT = os.environ.get("MINIO_INTERNAL_ADDRESS", "minio:9000")
+_DEFAULT_READ_ADDRESS = os.environ.get("MINIO_PUBLIC_ADDRESS", "http://minio:9000")
+_DEFAULT_BUCKET_NAME = os.environ.get("MINIO_BUCKET", "nv-ingest")
+
+EmbeddingStorageLoaderFactory = ModuleLoaderFactory(MODULE_NAME, MODULE_NAMESPACE, EmbeddingStorageModuleSchema)
+
+def upload_embeddings(df: pd.DataFrame, params: Dict[str, Any]) -> pd.DataFrame:
```

Comment:
We will definitely want to turn this into a multiprocessing stage. Anything doing uploads is likely to slow down the pipeline significantly.

Reply:
Can you link an example to what you mean? Like async? Or do you mean spawn multiple processes to do this. is there a maximum amount of data a single control message can hold ( I thought it would be limited by network packet limitations)? (still learning about capabilities here)